docs/extensions/guides/renderer-extension.md

```diff
@@ -374,20 +385,21 @@ The `feature` field must specify an instance which extends the abstract class `C
   abstract updateStatus(cluster: Cluster): Promise<ClusterFeatureStatus>;
 ```
 
-The `install()` method is typically called by Lens when a user has indicated that this feature is to be installed (i.e. clicked **Install** for the feature on the cluster settings page).
-The implementation of this method should install kubernetes resources using the `applyResources()` method, or by directly accessing the kubernetes api ([`K8sApi`](tbd)).
+The four methods listed above are defined as follows:
+
+* The `install()` method installs Kubernetes resources using the `applyResources()` method, or by directly accessing the [Kubernetes API](../api/README.md). Click **Install** on the cluster **Settings** page to call this method.
+
+* The `upgrade()` method upgrades the Kubernetes resources already installed, if they are relevant to the feature. Click **Upgrade** on the cluster **Settings** page to call this method.
+
+* The `uninstall()` method uninstalls Kubernetes resources using the [Kubernetes API](../api/README.md). Click **Uninstall** on the cluster **Settings** page to call this method.
```

Comment:
I'm not sure this is better. The extension developer must implement these functions. The extension developer should know that these functions get invoked when their end-user does the clicking.

Reply:
I've removed the instructions for what the end users should do, altogether. An alternative would be to keep the instructions, but prepend them with, "This feature is typically called when a user clicks..." Let me know if you prefer the latter – or something else!